src/concrete/ml/deployment/fhe_client_server.py

```diff
@@ -357,97 +388,78 @@ def get_serialized_evaluation_keys(self) -> bytes:
 
         # Decrypt the values
-        deserialized_decrypted_quantized_result = self.client.decrypt(
-            *deserialized_encrypted_quantized_result
-        )
+        q_result = self.client.decrypt(*q_result_enc)
 
-        return deserialized_decrypted_quantized_result
+        return q_result
 
+    # We should find a better name for `serialized_encrypted_quantized_result`
+    # FIXME: https://github.com/zama-ai/concrete-ml-internal/issues/4476
     def deserialize_decrypt_dequantize(
-        self, serialized_encrypted_quantized_result: Union[bytes, Tuple[bytes, ...]]
-    ) -> numpy.ndarray:
+        self, *serialized_encrypted_quantized_result: Optional[bytes]
+    ) -> Union[numpy.ndarray, Tuple[numpy.ndarray, ...]]:
         """Deserialize, decrypt and de-quantize the values.
 
         Args:
-            serialized_encrypted_quantized_result (Union[bytes, Tuple[bytes, ...]]): the
-                serialized, encrypted and quantized result
+            serialized_encrypted_quantized_result (Optional[bytes]): The serialized, encrypted and
+                quantized result
 
         Returns:
-            numpy.ndarray: the decrypted (de-quantized) values
+            Union[numpy.ndarray, Tuple[numpy.ndarray, ...]]: The clear float values.
         """
-        # Ensure the input is a tuple
-        serialized_encrypted_quantized_result = to_tuple(serialized_encrypted_quantized_result)
-
         # Decrypt and deserialize the values
-        deserialized_decrypted_quantized_result = self.deserialize_decrypt(
-            serialized_encrypted_quantized_result
-        )
-
-        deserialized_decrypted_quantized_result = to_tuple(deserialized_decrypted_quantized_result)
+        q_result = to_tuple(self.deserialize_decrypt(*serialized_encrypted_quantized_result))
 
         # De-quantize the values
-        deserialized_decrypted_dequantized_result = self.model.dequantize_output(
-            *deserialized_decrypted_quantized_result
-        )
-
-        deserialized_decrypted_dequantized_result = to_tuple(
-            deserialized_decrypted_dequantized_result
-        )
+        f_result = to_tuple(self.model.dequantize_output(*q_result))
 
         # Apply post-processing to the de-quantized values
-        deserialized_decrypted_dequantized_result = self.model.post_processing(
-            *deserialized_decrypted_dequantized_result
-        )
-
-        return deserialized_decrypted_dequantized_result
+        # Side note: `post_processing` method from built-in models (not Quantized Modules) only
+        # handles a single input. Calling the following is however not an issue for now as we expect
+        # 'f_result' to be a tuple of length 1 in this case anyway. Still, we need to make sure this
+        # does not break in the future if any built-in models starts to handle multiple outputs :
+        # FIXME: https://github.com/zama-ai/concrete-ml-internal/issues/4474
+        result = self.model.post_processing(*f_result)
```

Comment:
Can't we have an assert maybe? Either here or in built in post processing?

Reply:
I would still write explicitely in the variable name the difference between q_result f_result and result. I don't find these explicit enough.

Reply:
> Can't we have an assert maybe? Either here or in built in post processing?

I thought of it but I didn't find any proper way of doing it. The simplest way would be to check if `self.model` is a QuantizedModule (ok) or not (not ok), but I'm afraid this is a bit light and not very robust. Still would you prefer that we do that ? 

also there won't be a way to add any check in post-processing since the error is raised before entering in the method (ie: give several inputs instead of one)

Reply:
> I would still write explicitely in the variable name the difference between q_result f_result and result. I don't find these explicit enough.

yes I get this, but then I would be more in favor of a compromise then (like `float_result` or something) 😅 . I really found `deserialized_decrypted_dequantized_result` and associated hard to read and follow, especially when there are many successive lines. Like I couldn't differentiate `deserialized_decrypted_dequantized_result` from `deserialized_decrypted_quantized_result ` with a single look honestly 😅

Reply:
I added an assert with QM + I renamed the variables (`q` -> `quant`, `enc` -> `encrypted`)

Reply:
>  I really found deserialized_decrypted_dequantized_result and associated ones hard to read and follow

Agreed. Can you also make the before and after post processing variable different in

```python
result = to_tuple(self.model.dequantize_output(*result_quant))
...
result = self.model.post_processing(*result)
```

and then I am good !